app/src/main/kotlin/com/wire/android/ui/userprofile/image/AvatarPickerViewModel.kt

```diff
@@ -47,20 +51,23 @@ class AvatarPickerViewModel @Inject constructor(
     private fun loadAvatar() = viewModelScope.launch {
         try {
             dataStore.avatarAssetId.first()?.apply {
-                avatarRaw = (getUserAvatar(this) as PublicAssetResult.Success).asset
+                val avatarRaw = (getUserAvatar(this) as PublicAssetResult.Success).asset
+                val currentAvatarUri = getWritableAvatarUri(avatarRaw, context)
+                pictureState = PictureState.Initial(currentAvatarUri)
             }
         } catch (e: ClassCastException) {
             appLogger.e("There was an error loading the user avatar", e)
         }
     }
 
-    fun uploadNewPickedAvatarAndBack(imgUri: Uri, context: Context) {
+    fun uploadNewPickedAvatarAndBack(imgUri: Uri) {
         viewModelScope.launch {
             withContext(Dispatchers.IO) {
                 val data = imgUri.toByteArray(context)
                 val mimeType = imgUri.getMimeType(context) ?: DEFAULT_IMAGE_MIME_TYPE
                 val result = uploadUserAvatar(imageData = data)
                 if (result is UploadAvatarResult.Success) {
+                    getWritableAvatarUri(data, context)
```

Comment:
I think we can move functions like this to an object for example AvatarImageProcessor or AvatarImageManager and inject it into the viewmodel, that way we can also mock the object when unit testing the view model, but I already asked lots of changes in this PR so feel free to do it later if you feel like :P

Reply:
Done ! 👍